beamer/state_machine.py

```diff
@@ -217,22 +221,30 @@ def _handle_claim_made(event: ClaimMade, context: Context) -> HandlerResult:
     if claim.is_started:
         return False, None
 
+    events: list[Event] = []
+    if not claim.is_invalidated_l1_resolved and claim.invalidation_tx is not None:
+        events.append(
+            InitiateL1InvalidationEvent(
```

Comment:
Wouldn't it be possible to check here for L1 criteria fulfilled? In that way we wouldn't create an Event every iteration and then drop it again. 
This also counts for L1 resolution

Reply:
@palango What are your thoughts on that? We can also outsource it into its own PR as it was like that already for L1 resolution

Reply:
Yeah, let's do that in another PR. The problem is that we need to builds the infrastructure to emit events from the requests and claim handlers for this.